test/extensions/filters/http/jwt_authn/matcher_test.cc

```diff
@@ -176,6 +176,65 @@ TEST_F(MatcherTest, TestMatchConnectQuery) {
   EXPECT_FALSE(matcher->matches(headers));
 }
 
+TEST_F(MatcherTest, TestMatchPathSeparatedPrefix) {
+  const char config[] = R"(match:
+  path_separated_prefix: "/api")";
+  RequirementRule rule;
+  TestUtility::loadFromYaml(config, rule);
+  MatcherConstPtr matcher = Matcher::create(rule);
+  auto headers = TestRequestHeaderMapImpl{{":path", "/api"}};
+  EXPECT_TRUE(matcher->matches(headers));
```

Comment:
It would be good to use consistent test cases that are easy to correlate across the JWT and core router.

Reply:
Do you mean "equalize the test inputs per test case"?
I'll start with that, and if more work needs to be done - just elaborate on it.

Reply:
Not sure about the definition of equalize :) Anything you can do to make tests more consistent between JWT and core router. Since you're taking on #20578 I'm fine with punting to that if you prefer to do it there.

Reply:
I made it so the jwt test cases are a subset of the router ones, and the test strings are pretty much the same.
They don't look exactly the same, but the correlation should be obvious.